crates/sui-node/src/lib.rs

```diff
@@ -1591,10 +1603,22 @@ impl SuiNode {
                         &cur_epoch_store,
                         next_epoch_committee.clone(),
                         new_epoch_start_state,
-                        &checkpoint_executor,
+                        accumulator.clone(),
                     )
                     .await;
 
+                // No other components should be referencing state accumulator at this point.
+                // However, since checkpoint_service shutdown is asynchronous, we can't trust
+                // strong reference count. However it should still be safe to swap out state
+                // accumulator since we know we have already reached EndOfPublish
+                let new_accumulator = Arc::new(StateAccumulator::new(
```

Comment:
it might make more sense to use `Weak` in the places that we think shouldn't be holding old refs

Reply:
Done